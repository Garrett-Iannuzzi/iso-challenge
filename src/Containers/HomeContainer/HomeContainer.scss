form {
  border: solid 3px black;
  background-color:rgb(225, 207, 228);
  display: flex;
  flex-direction: column;
  opacity: .9;
  justify-content: space-evenly;
  height: 25em;
  margin: 3em;
  width: 30%;
  &:hover {
    opacity: 1;
    border-color: white;
  }
}

label {
  font-weight: bolder;
  font-size: 1.5em;
  margin: 1em;
}

input {
  padding-left: 10px;
  font-size: 1em;
  margin: 1em;
  height: 35px;
  width: 40%;
}

select {
  padding-left: 10px;
  margin: 1em;
  font-size: 1em;
  height: 35px;
  width: 40%;
}

.div-select-team {
  display: flex;
  justify-content: center;
}

.btn-start {
  font-size: 1em;
  width: 20%;
  font-weight: bolder;
  height: 35px;
  border-radius: 5px;
  margin-left: 40%;
  &:hover {
    cursor: pointer;
    background-color: rgb(225, 207, 228);
    color: white;
  }
}