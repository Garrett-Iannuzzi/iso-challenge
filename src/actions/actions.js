export const getTeamsInfo = (teamOne, teamTwo) => ({
  type: 'GET_TEAMS',
  teamOne,
  teamTwo
});

export const getPlayerInfo = (playerInfo) => ({
  type: 'GET_PLAYERS',
  playerInfo
});

export const getStatsInfoOne = (statsOne) => ({
  type: 'GET_STATS_ONE',
  statsOne
});

export const getStatsInfoTwo = (statsTwo) => ({
  type: 'GET_STATS_TWO',
  statsTwo
});

export const isLoggedIn = (isLoggedIn) => ({
  type: 'GET_LOGIN',
  isLoggedIn: isLoggedIn
})

export const selectPlayer = (player, team, index) => ({
  type: 'SELECT_PLAYER',
  player,
  team,
  index
})