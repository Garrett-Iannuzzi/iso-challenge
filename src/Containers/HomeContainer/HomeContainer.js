import React, { Component } from 'react';
import './HomeContainer.scss';
import Error from '../../Components/Error/Error';
import { connect } from 'react-redux';
import { getTeamsInfo, isLoggedIn } from '../../actions/actions';
import PropTypes from 'prop-types';

export class HomeContainer extends Component {
  constructor(props) {
    super(props)
    this.state ={
      teamOneName: '',
      teamTwoName: '',
      skillLevelOne: 'And-1',
      skillLevelTwo: 'And-1',
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  showError = () => {
    return(
      <Error history={this.props.history}/>
    )
  }

  submitTeams = () => {
    const {teamOneName, teamTwoName, skillLevelOne, skillLevelTwo } = this.state;
    const teamOne = { teamOneName, skillLevelOne, players: [] };
    const teamTwo = { teamTwoName, skillLevelTwo, players: [] };
    this.props.teamInfo(teamOne, teamTwo)
  }
  
  handleStartError = () => {
    const {teamOneName, teamTwoName, skillLevelOne, skillLevelTwo } = this.state;
    if (teamOneName === '' || teamTwoName === '' || skillLevelOne === '' || skillLevelTwo === '') {
      return this.props.isLoggedIn(false)
    } else {
      this.props.isLoggedIn(true)
      this.submitTeams()
      this.props.history.push('/game')
    }
  }
  

  render() {
    if(this.props.login === false) {
      return(this.showError())
    }
    return(
      <>   
        <div className='div-select-team'>
          <form>
            <label for='teamOneName'>Team 1 Name:</label>
            <input
              name='teamOneName'
              className='input-team-one'
              value={this.state.teamOneName}
              placeholder='Enter Team Name'
              type='text'
              maxLength='25'
              onChange={ (e) => this.handleChange(e) }
            />
            <label for='skillLevelOne'>Select Level:</label>
            <select
              id='skillLevelOne'
              value={this.state.skillLevelOne}
              name='skillLevelOne'
              className='select-skill-one'
              onChange={ (e) => this.handleChange(e) }
            >
              <option value='And-1'>And-1</option>
              <option value='D League'>D League</option>
              <option value='NBA'>NBA</option>
            </select>
          </form>
          <form>
            <label for='teamNameTwo'>Team 2 Name:</label>
            <input
              name='teamTwoName'
              className='input-team-two'
              value={this.state.teamTwoName}
              placeholder='Enter Team Name'
              type='text'
              maxLength='25'
              onChange={ (e) => this.handleChange(e) }
            />
            <label for='skillLevelTwo'>Select Level:</label>
            <select
              id='skillLevelTwo'
              value={this.state.skillLevelTwo}
              name='skillLevelTwo'
              className='select-skill-two'
              onChange={ (e) => this.handleChange(e) }
            >
              <option value='And-1'>And-1</option>
              <option value='D League'>D League</option>
              <option value='NBA'>NBA</option>
            </select>
          </form>
        </div>
        <button className='btn-start' type='button' onClick={ () => this.handleStartError() }>Start The Game</button>
      </>
    )
  }
}

export const mapStateToProps = state => ({
  login: state.isLoggedIn
})

export const mapDispatchToProps = dispatch => ({
  teamInfo: (teamOne, teamTwo) => dispatch(getTeamsInfo(teamOne, teamTwo)),
  isLoggedIn: (status) => dispatch(isLoggedIn(status))
})

export default connect(mapStateToProps, mapDispatchToProps)(HomeContainer)

HomeContainer.propTypes = {
  login: PropTypes.bool,
  isLoggedIn: PropTypes.func,
  teamInfo: PropTypes.func,
  history: PropTypes.func
}
