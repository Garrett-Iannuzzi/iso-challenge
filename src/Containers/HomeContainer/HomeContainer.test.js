import React from 'react';
import { HomeContainer, mapDispatchToProps } from './HomeContainer';
import { shallow } from 'enzyme';
import { teamInfo, getTeamsInfo } from '../../actions/actions';
import { mapDispatch } from '../HomeContainer/HomeContainer';

describe('HomeContainer', () => {

  let wrapper, mockEvent

  describe('homeContainer Component', () => {

    beforeEach(() => {
    wrapper = shallow(<HomeContainer
      teamInfo={jest.fn()}
    />)
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });

    it('Should setState when handleChange is called', () => {
      wrapper.setState({ teamOneName: '', teamTwoName: '', skillLevelOne: '', skillLevelTwo: '' });
      mockEvent = { target: {name: 'teamOneName', value:'Bazanga'} }
      wrapper.instance().handleChange(mockEvent);

      expect(wrapper.state()).toEqual({ teamOneName: 'Bazanga', teamTwoName: '', skillLevelOne: '', skillLevelTwo: '' })
    });

    it('Should call handleChange when teamOneName input is changed', () => {
      wrapper.instance().handleChange = jest.fn();
      const mockEvent = { target: {teamOneName: 'teamOneName', value:'Bazzzuca'} }

      wrapper.find('.input-team-one').simulate('change', mockEvent);

      expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent)
    });

    it('Should call handleChange when teamTwoName input is changed', () => {
      wrapper.instance().handleChange = jest.fn();
      const mockEvent = { target: {teamOneName: 'teamTwoName', value:'Bazzzuca'} }

      wrapper.find('.input-team-two').simulate('change', mockEvent);

      expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent)
    });

    it('Should call handleChange when player one selects skill level', () => {
      wrapper.instance().handleChange = jest.fn();
      const mockEvent = { target: { skillLevelOne: 'skillLevelOne', value:'NBA' } }

      wrapper.find('.select-skill-one').simulate('change', mockEvent);

      expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent)
    });

    it('Should call handleChange when player two selects skill level', () => {
      wrapper.instance().handleChange = jest.fn();
      const mockEvent = { target: { skillLevelTwo: 'skillLevelTwo', value:'NBA' } }

      wrapper.find('.select-skill-two').simulate('change', mockEvent);

      expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent)
    });
  });

  describe('mapDispatchToProps', () => {

    it('Should call mapDispatchToProps with correct team info when handleStartError is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getTeamsInfo({ teamInfo: [{ teamOneName: 'G-Swish', skillLevelOne: 'NBA' }, { teamTwoName: 'D-Buckets', skillLevelTwo: 'NBA' }] });

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.teamInfo({ teamInfo: [{ teamOneName: 'G-Swish', skillLevelOne: 'NBA' }, { teamTwoName: 'D-Buckets', skillLevelTwo: 'NBA' }] });

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  });

})