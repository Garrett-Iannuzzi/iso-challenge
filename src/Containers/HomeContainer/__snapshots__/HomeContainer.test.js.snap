// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer homeContainer Component should match the snapshot 1`] = `
<Fragment>
  <div
    className="div-select-team"
  >
    <form>
      <label
        for="teamOneName"
      >
        Team 1 Name:
      </label>
      <input
        className="input-team-one"
        maxLength="25"
        name="teamOneName"
        onChange={[Function]}
        placeholder="Enter Team Name"
        type="text"
        value=""
      />
      <label
        for="skillLevelOne"
      >
        Select Level:
      </label>
      <select
        className="select-skill-one"
        id="skillLevelOne"
        name="skillLevelOne"
        onChange={[Function]}
        value="And-1"
      >
        <option
          value="And-1"
        >
          And-1
        </option>
        <option
          value="D League"
        >
          D League
        </option>
        <option
          value="NBA"
        >
          NBA
        </option>
      </select>
    </form>
    <form>
      <label
        for="teamNameTwo"
      >
        Team 2 Name:
      </label>
      <input
        className="input-team-two"
        maxLength="25"
        name="teamTwoName"
        onChange={[Function]}
        placeholder="Enter Team Name"
        type="text"
        value=""
      />
      <label
        for="skillLevelTwo"
      >
        Select Level:
      </label>
      <select
        className="select-skill-two"
        id="skillLevelTwo"
        name="skillLevelTwo"
        onChange={[Function]}
        value="And-1"
      >
        <option
          value="And-1"
        >
          And-1
        </option>
        <option
          value="D League"
        >
          D League
        </option>
        <option
          value="NBA"
        >
          NBA
        </option>
      </select>
    </form>
  </div>
  <button
    className="btn-start"
    onClick={[Function]}
    type="button"
  >
    Start The Game
  </button>
</Fragment>
`;
